runtime/ops/ops_buffer.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::AnyError;
+use deno_core::futures::future::FutureExt;
+use deno_core::BufVec;
+use deno_core::Op;
+use deno_core::OpFn;
+use deno_core::OpState;
+use deno_core::ZeroCopyBuf;
+use std::boxed::Box;
+use std::cell::RefCell;
+use std::future::Future;
+use std::rc::Rc;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct RequestHeader {
+  pub promise_id: u32,
+  pub argument: u32,
+}
+
+impl RequestHeader {
+  pub fn from_raw(bytes: &[u8]) -> Option<Self> {
+    if bytes.len() < 2 * 4 {
+      return None;
+    }
+    let ptr = bytes.as_ptr() as *const u32;
+    let ints = unsafe { std::slice::from_raw_parts(ptr, 2) };
+    Some(Self {
+      promise_id: ints[0],
+      argument: ints[1],
+    })
+  }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ResponseHeader {
+  pub promise_id: u32,
```

Comment:
I hadn't realized this either, but I guess we're using 64 bit promises now.

Reply:
Changed it to 64 bits.